runtime/native/unsafe.cpp

```diff
@@ -277,9 +287,25 @@ Java_sun_misc_Unsafe_staticFieldOffset(JNIEnv *env, jobject, jobject) {
     UnsafeUnimplemented("Java_sun_misc_Unsafe_staticFieldOffset");
 }
 
+unsigned int fieldSlotOffset = -1;
```

Comment:
A global variable here seems unsafe since this will be shared across the environment for every single class.

Was this here for debugging purposes?

Reply:
So this is just to cache the offset into field for the slot variable. Since the field object never changes, it should be fine being a global variable.

Reply:
Ah ok I see, right. Well, in that case you should probably update fieldSlotOffset :) as is it never gets written and stays -1 forever